pkg/analyzer/detection/detection.go

```diff
@@ -99,6 +116,65 @@ func detectAndAnnotateCertifiedRHELComponents(name string, files analyzer.Files,
 	return rhelfeatures, languageComponents, nil
 }
 
+// getHardcodedRHCOSContentSets returns hard coded content sets for supported
+// RHCOS versions. When the RHCOS version is not supported, it returns empty. The
+// function assumes we are working in a RHCOS namespace, and uses
+// `/etc/os-release` to gather versions and metadata. But, if the expected
+// metadata is not found, is inconsistent or cannot be retrieved, it returns an
+// error.
+func getHardcodedRHCOSContentSets(files analyzer.Files) ([]string, error) {
+	// Get metadata from etc/os-release, ensure it exists.
+	osReleaseData, exists := files.Get("etc/os-release")
+	if !exists {
+		return nil, fmt.Errorf("etc/os-release not found")
+	}
+	keys := []string{"ID", "OPENSHIFT_VERSION", "RHEL_VERSION", "VERSION_ID"}
+	metadata := osrelease.GetOSReleaseMap(osReleaseData.Contents, keys...)
+	for _, k := range keys {
+		if _, ok := metadata[k]; !ok {
+			return nil, fmt.Errorf("not a RHCOS namespace: missing %s in etc/os-release", k)
+		}
+	}
+	if metadata["ID"] != "rhcos" {
+		return nil, fmt.Errorf("not a RHCOS namespace: ID is %v", metadata["ID"])
+	}
+	// Verify the RHCOS version is supported.
+	ok, err := isRPMVersionInInterval(metadata["VERSION_ID"], "4.7", "4.10")
+	if !ok {
+		return nil, err
+	}
+	// Format the content sets based on the metadata found.
+	rhelSuffix := strings.Replace(metadata["RHEL_VERSION"], ".", "_DOT_", 1)
+	return []string{
+		// The mapping from RHCOS version to RHEL minor releases can be found here:
+		// https://access.redhat.com/articles/6907891
```

Comment:
Out of interest: This gives us the version mappings, but how would I get from version mappings to the content set names in this slice? Do we have a resource for that as well or is that tribal knowledge?

Reply:
Could you help me understand your question? Would you like to know how we formed the content set strings?

If yes, this was confirmed with RHCOS for 4.7, 4.8, and 4.9 -- The release suffix was confirmed by the table referenced and inspecting https://www.redhat.com/security/data/metrics/repository-to-cpe.json

Reply:
I was missing the JSON you have provided here in my mental model, thank you!